shared/src/ledger/eth_bridge/vp/mod.rs

```diff
@@ -67,6 +124,12 @@ where
             verifiers_len = verifiers.len(),
             "Ethereum Bridge VP triggered",
         );
+
+        // first check if Nam is being escrowed
+        if keys_changed.contains(&balance_key(&xan(), &super::ADDRESS)) {
+            return self.check_escrow(verifiers);
+        }
```

Comment:
```suggestion
        if keys_changed.contains(&balance_key(&xan(), &super::ADDRESS))
            && !keys_changed
                .iter()
                .any(|key| storage::is_eth_bridge_key(key))
        {
            return self.check_escrow(verifiers);
        }
```

Re. https://github.com/anoma/namada/pull/703#discussion_r1013022043, looking again at this, if NAM is sent to the Eth bridge, the Eth bridge VP passes iff the bridge pool is in the verifiers list. We want to check no Eth bridge keys changed at all as well in that case, maybe something like this.

Reply:
I actually changed the extract keys function to make the dual functions of the VP explicit.